client/payment-details/summary/missing-order-notice/index.tsx

```diff
@@ -0,0 +1,149 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { Button, RadioControl } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import Loadable from 'wcpay/components/loadable';
+import { useState } from '@wordpress/element';
+import ConfirmationModal from 'wcpay/components/confirmation-modal';
+import interpolateComponents from '@automattic/interpolate-components';
+
+/**
+ * Internal dependencies.
+ */
+
+import './style.scss';
+import CardNotice from 'wcpay/components/card-notice';
+
+interface MissingOrderNoticeProps {
+	isLoading: boolean;
+	formattedAmount: string;
+}
+
+const MissingOrderNotice: React.FC< MissingOrderNoticeProps > = ( {
+	isLoading,
+	formattedAmount,
+} ) => {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+
+	const [ reason, setReason ] = useState< string | null >( null );
+
+	const handleOnButtonClick = () => {
+		setIsModalOpen( true );
+	};
+
+	const handleModalCancel = () => {
+		setIsModalOpen( false );
+	};
+
+	const handleModalConfirmation = () => {
+		// TODO: Handle the refund.
+	};
+
+	return (
+		<>
+			<Loadable isLoading={ isLoading } placeholder="">
+				<CardNotice
+					actions={
+						<Button
+							isPrimary={ true }
+							isSmall={ false }
+							onClick={ handleOnButtonClick }
+						>
+							{ __( 'Refund', 'woocommerce-payments' ) }
+						</Button>
+					}
+				>
+					{ __(
+						'This transaction is not connected to order. Investigate this purchase and refund the transaction as needed.',
+						'woocommerce-payments'
+					) }
+				</CardNotice>
+			</Loadable>
+			{ isModalOpen && (
+				<ConfirmationModal
+					title={ __( 'Refund Transaction', 'woocommerce-payments' ) }
+					actions={
+						<>
+							<Button onClick={ handleModalCancel } isSecondary>
+								{ __( 'Cancel', 'woocommerce-payments' ) }
+							</Button>
+							<Button
+								onClick={ handleModalConfirmation }
+								isPrimary
+							>
+								{ __(
+									'Refund transaction',
+									'woocommerce-payments'
+								) }
+							</Button>
+						</>
+					}
+					onRequestClose={ handleModalCancel }
+				>
+					<p>
+						{ interpolateComponents( {
+							mixedString: sprintf(
+								__(
+									'This will issue a full refund of {{strong}}%s{{/strong}} to the customer.',
+									'woocommerce-payments'
+								),
+								formattedAmount
+							),
+							components: {
+								strong: <strong />,
+							},
+						} ) }
+					</p>
+					<RadioControl
+						className="missing-order-notice__modal__reason"
```

Comment:
This class name deviates a bit from the BEM convention. Usually you'll only see "__" used once to denote an Element that belongs to a Block, but in this case we have two. I'd say that the modal is Block, and then we have reason element. Something like:

```suggestion
						className="missing-order-notice-modal__reason"
```